components/buildless-serverless/api/v1alpha2/function_types.go

```diff
@@ -0,0 +1,192 @@
+	// +kubebuilder:validation:Required
+	Source Source `json:"source"`
+
+	// Specifies an array of key-value pairs to be used as environment variables for the Function.
+	// You can define values as static strings or reference values from ConfigMaps or Secrets.
+	// For configuration details, see the [official Kubernetes documentation](https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/).
+	// +kubebuilder:validation:XValidation:message="Following envs are reserved and cannot be used: ['FUNC_RUNTIME','FUNC_HANDLER','FUNC_PORT','FUNC_HANDLER_SOURCE','FUNC_HANDLER_DEPENDENCIES','MOD_NAME','NODE_PATH','PYTHONPATH']",rule="(self.all(e, !(e.name in ['FUNC_RUNTIME','FUNC_HANDLER','FUNC_PORT','MOD_NAME','NODE_PATH','PYTHONPATH'])))"
+	Env []corev1.EnvVar `json:"env,omitempty"`
+
+	// Specifies resources requested by the Function and the build Job.
+	// +optional
+	ResourceConfiguration *ResourceConfiguration `json:"resourceConfiguration,omitempty"`
+
+	// Defines the exact number of Function's Pods to run at a time.
+	// If  the Function is targeted by an external scaler,
+	// then the **Replicas** field is used by the relevant HorizontalPodAutoscaler to control the number of active replicas.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:default:=1
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
+
+	// Specifies Secrets to mount into the Function's container filesystem.
+	SecretMounts []SecretMount `json:"secretMounts,omitempty"`
+}
+
+type Source struct {
+	// Defines the Function as git-sourced. Can't be used together with **Inline**.
+	// +optional
+	//	GitRepository *GitRepositorySource `json:"gitRepository,omitempty"`
+
+	// Defines the Function as the inline Function. Can't be used together with **GitRepository**.
+	// +optional
+	Inline *InlineSource `json:"inline,omitempty"`
+}
+
+type InlineSource struct {
+	// Specifies the Function's full source code.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Source string `json:"source"`
+
+	// Specifies the Function's dependencies.
+	//+optional
+	Dependencies string `json:"dependencies,omitempty"`
+}
+
+type ResourceConfiguration struct {
+	// Deprecated: Specifies resources requested by the build Job's Pod.
+	// This setting should be removed from a future version where Functions won't require building images.
+	// +optional
+	// +kubebuilder:validation:XValidation:message="Use profile or resources",rule="has(self.profile) && !has(self.resources) || !has(self.profile) && has(self.resources)"
+	// +kubebuilder:validation:XValidation:message="Invalid profile, please use one of: ['local-dev','slow','normal','fast']",rule="(!has(self.profile) || self.profile in ['local-dev','slow','normal','fast'])"
+	Build *ResourceRequirements `json:"build,omitempty"`
```

Comment:
There will be no build phase so we can get rid of this field. But it's a suggestion only and it can be done later :)

Reply:
You are right, I've deleted it